### ISSUE TITLE
Custom fields with capital letters do not work as GraphQL filters

### ISSUE THREAD

**Issue by ozmusic** [nautobot/nautobot#1275]
### Environment
* Python version: 3.8.11
* Nautobot version: 1.2.3

A custom field with capital letters in the slug does not work as a filter in GraphQL

### Steps to Reproduce
1. Create a custom field with a capital letter
    (e.g. QL_Test, dcim | site, Text, exact filter logic) 
2. Use the custom field somewhere
3. Create a graphql query to filter based on it
```
  query {
    sites (cf_ql_test: "test1") {
      name
    }
  }
```

Only the site matching the query is returned

All sites are returned

**Comment by bryanculver** [comment]
Thanks for the submission @ozmusic. Upon review this will likely be a breaking change as it affect more than just the GraphQL query, and will deal with normalization under the hood.

**Comment by glennmatthews** [comment]
Related to #824.

**Comment by lampwins** [comment]
This will be accounted for and fixed by #824, so closing this one.
